### PR TITLE
[SPARK-45085][SQL] Merge UNSUPPORTED_TEMP_VIEW_OPERATION into UNSUPPORTED_VIEW_OPERATION and refactor some logic

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to:
- Merge UNSUPPORTED_TEMP_VIEW_OPERATION into UNSUPPORTED_VIEW_OPERATION in error classes.
- Refactor some code.
- Fix some issue about UT.

### Why are the changes needed?
- The changes improve the error framework.
- Make code clear.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
- Pass GA.
- Manual test.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
Waiting: https://github.com/apache/spark/pull/42830

**Comment by panbingkun** [comment]
cc @cloud-fan @MaxGekk

File: sql/core/src/test/resources/sql-tests/analyzer-results/change-column.sql.out [hunk 1]
```diff
@@ -206,10 +206,10 @@ ALTER TABLE temp_view CHANGE a TYPE INT
 -- !query analysis
 org.apache.spark.sql.AnalysisException
 {
-  "errorClass" : "UNSUPPORTED_TEMP_VIEW_OPERATION.WITH_SUGGESTION",
+  "errorClass" : "UNSUPPORTED_VIEW_OPERATION.WITHOUT_SUGGESTION",
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/analyzer-results/change-column.sql.out:209]: to double-check if we are symmetrical: what happens if we use ALTER VIEW to alter a table?

Comment (panbingkun) [reply sql/core/src/test/resources/sql-tests/analyzer-results/change-column.sql.out:209]: Let me to check it.

Comment (panbingkun) [reply sql/core/src/test/resources/sql-tests/analyzer-results/change-column.sql.out:209]: image
image

Comment (panbingkun) [reply sql/core/src/test/resources/sql-tests/analyzer-results/change-column.sql.out:209]: From the test results, it is reasonable and symmetrical.

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
@cloud-fan Are you ok with the changes?

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

**Comment by cloud-fan** [comment]
late LGTM

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/v2ResolutionPlans.scala [hunk 2]
```diff
@@ -44,7 +44,7 @@ case class UnresolvedNamespace(multipartIdentifier: Seq[String]) extends Unresol
 case class UnresolvedTable(
     multipartIdentifier: Seq[String],
     commandName: String,
-    hint: Boolean = false) extends UnresolvedLeafNode
+    suggestAlternative: Boolean = false) extends UnresolvedLeafNode
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/v2ResolutionPlans.scala:47]: shall we also update the parameter name in `AstBuilder#createUnresolvedTable`?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/v2ResolutionPlans.scala [hunk 3]
```diff
@@ -54,7 +54,7 @@ case class UnresolvedView(
     multipartIdentifier: Seq[String],
     commandName: String,
     allowTemp: Boolean,
-    hint: Boolean) extends UnresolvedLeafNode
+    suggestAlternative: Boolean) extends UnresolvedLeafNode
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/v2ResolutionPlans.scala:57]: to be consistent with `UnresolvedTable`, shall we give it a default value (false) as well?

Comment (panbingkun) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/v2ResolutionPlans.scala:57]: The above modifications are submitted as following PR, and I will do it today.

Comment (panbingkun) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/v2ResolutionPlans.scala:57]: A new follow up pr: https://github.com/apache/spark/pull/42917.

**Pull request closed without merging**